File: src/src/Subsystems/Drivetrain.cpp
#include "Drivetrain.h"
#include <Joystick.h>
#include <VictorSP.h>
#include "../RobotMap.h"
#include <SmartDashboard/SmartDashboard.h>

Drivetrain::Drivetrain() :
		frc::Subsystem("Drivetrain") {
	leftMotor = new VictorSP(leftDriveMotor);
	rightMotor = new VictorSP(rightDriveMotor);
	leftMotor->SetInverted(true);
}

void Drivetrain::InitDefaultCommand() {
	// Set the default command for a subsystem here.
	// SetDefaultCommand(new MySpecialCommand());
}

// Put methods for controlling this subsystem
// here. Call these from Commands.
void Drivetrain::Set(double leftValue, double rightValue){
	if (!IsEnabled){
		leftValue = 0;
		rightValue = 0;
	}
	leftMotor->Set(leftValue);
	rightMotor->Set(rightValue);
	frc::SmartDashboard::PutNumber("LeftMotor", leftValue);
	frc::SmartDashboard::PutNumber("RightMotor", rightValue);
	
}

void Drivetrain::Disable(){
	Set(0, 0);
	IsEnabled = false;
}

void Drivetrain::Enable(){
	IsEnabled = true;
}

File: src/org/usfirst/frc/team9159/robot/subsystems/Drivetrain.java
package org.usfirst.frc.team9159.robot.subsystems;

import edu.wpi.first.wpilibj.VictorSP;
import edu.wpi.first.wpilibj.command.Subsystem;
import org.usfirst.frc.team9159.robot.RobotMap;
import org.usfirst.frc.team9159.robot.commands.TeleopDrive;

/**
 *
 */
public class Drivetrain extends Subsystem {
	private VictorSP leftMotor;
	private VictorSP rightMotor;
	
	// Put methods for controlling this subsystem
    // here. Call these from Commands.
	
	public Drivetrain(){
		leftMotor = new VictorSP(RobotMap.LEFT_MOTOR);
		rightMotor = new VictorSP(RobotMap.RIGHT_MOTOR);	
	}
	public void setRaw(double left, double right){
			leftMotor.set(left);
			rightMotor.set(-right);
	}

    public void initDefaultCommand() {
        // Set the default command for a subsystem here.
        //setDefaultCommand(new MySpecialCommand());
    	setDefaultCommand(new TeleopDrive());
    }
}

